File: vbnet/App_Code/Modules/menu/MenuModule.vb

Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Mvc

Partial Public Class WebExModel
    Private _menu As IEnumerable(Of MenuItem)
    <LoadModel> _
    Public Sub LoadMenu(ctrl As Controller)
        _menu = {New MenuItem() With { _
            .Name = "Main", _
            .Url = ctrl.Url.Action("index", New With { _
                .page = "" _
            }) _
        }, New MenuItem() With { _
            .Name = "About", _
            .Url = ctrl.Url.Action("index", New With { _
                 .page = "about" _
            }) _
        }}
    End Sub
    <[Module]> _
    Public ReadOnly Property Menu() As IEnumerable(Of MenuItem)
        Get
            Return _menu
        End Get
    End Property
End Class

Public Class MenuItem
    Public Property Name() As String
        Get
            Return m_Name
        End Get
        Set(value As String)
            m_Name = Value
        End Set
    End Property
    Private m_Name As String
    Public Property Url() As String
        Get
            Return m_Url
        End Get
        Set(value As String)
            m_Url = Value
        End Set
    End Property
    Private m_Url As String
End Class

File: packages/WebEx.Menu.1.0.2/content/App_Code/WebEx.Modules/menu/MenuModule.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Mvc;
using WebEx.Core;

public class MenuModule : IModuleWithModel
{
    private IEnumerable<MenuItem> _menu;
    public MenuModule(Controller ctrl)
    {
        _menu = LoadMenu(ctrl);
    }
    public static IEnumerable<MenuItem> LoadMenu(Controller ctrl)
    {
        return new[] { 
            new MenuItem {Name = "Main", Url=ctrl.Url.Action("index", new {page=""})},
            new MenuItem {Name = "About", Url=ctrl.Url.Action("index", new {page="about"})}
        };
    }

    public IEnumerable<MenuItem> Menu
    {
        get
        {
            return _menu;
        }
    }
    public object Model
    {
        get
        {
            return Menu;
        }
    }

    public IModuleView GetView(string type, HtmlHelper html)
    {
        return new ModuleAutoView(type);
    }
}

public class MenuItem
{
    public string Name { get; set; }
    public string Url { get; set; }
}